### PR TITLE
Domino Loss Functions

### PR DESCRIPTION
**Pull request by coreyjadams**
This commit creates alternative versions of the domino loss functions that are significantly simpler and shorter, while producing numerically consistent results.

Original functions are maintained in this commit and the training script compares individual loss components as well as total loss.

This commit also ensure `area` has the right dimension in the train and validation loop.  It's unchanged in the data sources or anywhere else (the model unsqueezes it too).  That is still a general issue to resolve, but the loss should be correct here.

# PhysicsNeMo Pull Request

## Checklist

- [ ] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/physicsnemo/blob/main/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/physicsnemo/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/physicsnemo/issues) is linked to this pull request.

### CODE REVIEW

**Comment by coreyjadams** [comment]
And, here's a snippet of the training w/ `loss_type: "mse"`

```
[2025-04-23 07:58:07,945][Train][INFO] - loss_fn_vol agreement: True
[2025-04-23 07:58:07,948][Train][INFO] - loss_fn_surface agreement: True
[2025-04-23 07:58:07,948][Train][INFO] - loss_fn_area agreement: True
[2025-04-23 07:58:07,949][Train][INFO] - total loss agreement: True
[2025-04-23 07:58:08,470][Train][INFO] - Device cuda:0, batch processed: 9
  loss volume: 0.02605
  loss surface: 0.02330
  loss integral: 0.00002
  loss surface area: 0.00056
  GPU memory used: 48.79901123046875 Gb
  GPU memory delta: 0.0 Gb

[2025-04-23 07:58:11,881][Train][INFO] - loss_fn_vol agreement: True
[2025-04-23 07:58:11,884][Train][INFO] - loss_fn_surface agreement: True
[2025-04-23 07:58:11,884][Train][INFO] - loss_fn_area agreement: True
[2025-04-23 07:58:11,886][Train][INFO] - total loss agreement: True
[2025-04-23 07:58:12,401][Train][INFO] - Device cuda:0, batch processed: 10
  loss volume: 0.03133
  loss surface: 0.02049
  loss integral: 0.00000
  loss surface area: 0.00048
  GPU memory used: 48.79901123046875 Gb
  GPU memory delta: 0.0 Gb

[2025-04-23 07:58:16,082][Train][INFO] - loss_fn_vol agreement: True
[2025-04-23 07:58:16,086][Train][INFO] - loss_fn_surface agreement: True
[2025-04-23 07:58:16,086][Train][INFO] - loss_fn_area agreement: True
[2025-04-23 07:58:16,087][Train][INFO] - total loss agreement: True
[2025-04-23 07:58:16,611][Train][INFO] - Device cuda:0, batch processed: 11
  loss volume: 0.03989
  loss surface: 0.02423
  loss integral: 0.00000
  loss surface area: 0.00051
  GPU memory used: 48.49237060546875 Gb
  GPU memory delta: -0.306640625 Gb

[2025-04-23 07:58:16,612][Train][INFO] -  Device cuda:0,  batch: 11, loss norm: 0.05226
[2025-04-23 07:58:16,616][Train][INFO] - Device cuda:0, Epoch 1 took 45.99729591188952 seconds
[2025-04-23 07:58:20,023][Train][INFO] - loss_fn_vol agreement: True
[2025-04-23 07:58:20,025][Train][INFO] - loss_fn_surface agreement: True
[2025-04-23 07:58:20,026][Train][INFO] - loss_fn_area agreement: True
[2025-04-23 07:58:20,026][Train][INFO] - total loss agreement: True
[2025-04-23 07:58:23,324][Train][INFO] - loss_fn_vol agreement: True
[2025-04-23 07:58:23,328][Train][INFO] - loss_fn_surface agreement: True
[2025-04-23 07:58:23,328][Train][INFO] - loss_fn_area agreement: True
[2025-04-23 07:58:23,329][Train][INFO] - total loss agreement: True
[2025-04-23 07:58:23,330][Train][INFO] - Device cuda:0 LOSS train 0.08511 valid 0.05631 Current lr 0.001Integral factor 100
Device cuda:0, Best val loss 0.0563129223883152, Time taken 52.71513529098593
```

**Review by peterdsharpe — APPROVED** [review]
Overall, a massive improvement to readability, and looks good to merge to me.

Longer-term, there are a few possible correctness footguns that jump out while reading `train.py` - I think we should keep an eye on for future PRs. These don't have to be addressed in this PR, and they were pre-existing before this PR - so I don't think this should delay merging this PR. These include:

- Defining an placeholder `padded_value=-10` rather than something like `np.nan`
- Hard-coding `vel_inlet=30.0` in the lift- and drag-loss-functions
- Hard-coding `scale_factor=1.0` in the area-based loss function

We also might want to add the ability to control the relative weights of loss components in `total_loss_terms` in `train_epoch()`, since currently this is arbitrarily set - on different datasets, or with different goals, the best weights might be different.
